tools/nl/loadtest/load.py

```diff
@@ -0,0 +1,157 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import concurrent.futures
+import csv
+import json
+
+from absl import app
+from absl import flags
+import requests
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_integer('parallel_requests', 0,
+                     'Number of requests to run in parallel')
+flags.DEFINE_integer('total_requests', 0, 'Total number of requests to run')
+
+_OUTPUT_FILE = 'load_results.json'
+_QUERY_FILE = 'queryset.csv'
+_URL = 'https://bard.datacommons.org/nodejs/query?apikey=s5URv2M1vU3dfOjlDRHQj99fvvn8uQzqWLGxTrxOXBjGaTEZ&q='
+_RESULT_KEY_CODE = 'code'
+_RESULT_KEY_ERROR_CONTENT = 'errorContent'
+_RESULT_KEY_NL_TIME = 'nlTime'
+_RESULT_KEY_TOTAL_TIME = 'totalTime'
+_RESULT_KEY_TILE_TIME = 'tileTime'
+session = requests.Session()
+
+
+# Runs a single query and returns the response code.
+def _run_query(query):
+  # Make the API request
+  print(f'Running query: {query}')
+  try:
+    response = session.get(_URL + query, timeout=None)
+    # response = session.get(_URL, timeout=None)
+    if response.status_code == 200:
+      debug_timing = response.json().get('debug', {}).get('timing', {})
+      return {
+          _RESULT_KEY_CODE: str(response.status_code),
+          _RESULT_KEY_NL_TIME: debug_timing.get('getNlResult'),
+          _RESULT_KEY_TOTAL_TIME: debug_timing.get('total'),
+          _RESULT_KEY_TILE_TIME: debug_timing.get('getTileResults')
+      }
+    return {
+        _RESULT_KEY_CODE: str(response.status_code),
+        _RESULT_KEY_ERROR_CONTENT: str(response.content)
+    }
+  except Exception as e:
+    return {_RESULT_KEY_CODE: 'Exception', _RESULT_KEY_ERROR_CONTENT: str(e)}
+
+
+# Get a list of queries
+def _get_query_list():
+  query_list = []
+  with open(_QUERY_FILE) as inf:
+    for row in csv.reader(inf):
+      if not row:
+        continue
+      query = row[0].strip()
+      if not query or query.startswith('#') or query.startswith('//'):
+        continue
+      assert ';' not in query, 'Multiple query not yet supported'
+      query_list.append(query)
+  return query_list
+
+
+# Take the list of results from running the queries and get the result output
+# object.
+def _get_result_output(run_results):
+  response_codes = {}
+  times_seen = {}
+  result = {
+      _RESULT_KEY_NL_TIME: {
+          'avg': 0,
+          'max': 0,
+          'min': 0
+      },
+      _RESULT_KEY_TILE_TIME: {
+          'avg': 0,
+          'max': 0,
+          'min': 0
+      },
+      _RESULT_KEY_TOTAL_TIME: {
+          'avg': 0,
+          'max': 0,
+          'min': 0
+      },
+  }
+  error_content = {}
+  for r in run_results:
+    code = r.get('code')
```

Comment:
use the `_REQUEST_KEY_...` constants here?

Reply:
done